File: fvc-frontend/src/types/location.ts
export interface LocationDto {
  id: string;
  name: string;
  address?: string;
  capacityDefault?: number | null;
  description?: string;
  lat?: number | null;
  lng?: number | null;
  isActive: boolean;
  createdBy?: { id: string; fullName?: string } | null;
  updatedBy?: { id: string; fullName?: string } | null;
  createdAt?: string;
  updatedAt?: string;
}

export interface LocationCreateRequest {
  name: string;
  address?: string;
  capacityDefault?: number;
  description?: string;
  lat?: number;
  lng?: number;
}

export interface LocationUpdateRequest {
  name?: string;
  address?: string;
  capacityDefault?: number;
  description?: string;
  lat?: number;
  lng?: number;
  isActive?: boolean;
}

export interface LocationFilters {
  page?: number;
  size?: number;
  search?: string;
  isActive?: boolean;
}

File: fvc-api/src/main/java/sep490g65/fvcapi/entity/Location.java
package sep490g65.fvcapi.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "locations",
        uniqueConstraints = @UniqueConstraint(name = "uk_location_name", columnNames = "name")
)
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Location extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false, length = 150)
    private String name;

    @Column(length = 255)
    private String address;

    @Column(name = "capacity_default")
    private Integer capacityDefault;

    @Column(length = 500)
    private String description;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @Column(precision = 10, scale = 7)
    private BigDecimal lat;

    @Column(precision = 10, scale = 7)
    private BigDecimal lng;

    @ManyToOne(optional = false)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @ManyToOne
    @JoinColumn(name = "updated_by")
    private User updatedBy;
}


